src/Bbt.Maybe/Maybe.cs

```diff
@@ -0,0 +1,243 @@
+﻿namespace Bbt.Maybe
+{
+    using System;
+    using System.Globalization;
+    using System.Runtime.Serialization;
+
+    /// <summary>
+    /// Represents a reference type which may be null or not null.
+    /// Implementation of the "maybe monad" pattern / Optional-pattern.
+    /// https://smellegantcode.wordpress.com/2008/12/11/the-maybe-monad-in-c/.
+    /// </summary>
+    /// <typeparam name="T">The reference type.</typeparam>
+    [Serializable]
+    public struct Maybe<T> : ISerializable, IEquatable<Maybe<T>>
+        where T : class
+    {
+        private T mValue;
```

Comment:
We normally don't prefix variables in our GitHub projects (so this would become `value`). But can also be cleaned up in a subsequent PR

Reply:
Created #13 for tracking this